nyx_client/nyx_client/client.py

```diff
@@ -599,20 +609,25 @@ def create_data(
             "lang": lang,
             "status": status,
             "preview": preview_base64_string,
-            "downloadURL": download_url,
             "contentType": content_type,
         }
+        if download_url:
+            data["download_url"] = download_url
         if price:
             data["price"] = price
-
         if license_url:
             data["licenseURL"] = license_url
 
-        multipart_data = MultipartEncoder(
-            fields={
-                "productMetadata": json.dumps(data),
-            }
-        )
+        fields = {
+            "productMetadata": json.dumps(data),
+        }
+
+        if file:
+            file.seek(0)
+            # Add file to fields with filename
+            fields["productData"] = (name, file, content_type)
```

Comment:
Does this work as expected?
[Here](https://github.com/urllib3/urllib3/blob/2.2.3/src/urllib3/filepost.py#L73-L81), where field data is written - it looks like it will only accept str & bytes instances? :thinking: 

Also worth noting (but nothing we can do anything about with this approach) is that the whole file [will be read into memory first](https://github.com/urllib3/urllib3/blob/2.2.3/src/urllib3/filepost.py#L65).
(The way to solve that later, if necessary, is to not use multi-part or to write that bit ourselves.)